### PR TITLE
Update stacking_weights_crps.R

### PR DESCRIPTION
**Pull request by seabbs**
Syntax and imports

### CODE REVIEW

File: R/stacking_weights_crps.R [hunk 1]
```diff
@@ -42,20 +42,20 @@
 #' \dontrun{
 #' library(data.table)
 #' splitdate <- as.Date("2020-03-28")
-#' data <- stackr::example_data 
+#' data <- data.table::setDT(stackr::example_data)
 #' 
-#' traindata <- data[date <= splitdate, ]
-#' testdata <- data[date > splitdate, ]
+#' traindata <- data[date <= splitdata]
```
Comment (nikosbosse) [inline R/stacking_weights_crps.R]: this should be splitdate I think instead of splitdata

File: R/stacking_weights_crps.R [hunk 2]
```diff
@@ -32,7 +32,7 @@
 #' 
 #' @param dirichlet_alpha prior for the weights. Default is 1.001
 #' 
-#' @import data.table
+#' @importFrom data.table setDT
```
Comment (nikosbosse) [inline R/stacking_weights_crps.R:35]: can you add rbindlist please?
the reason I imported the whole data.table is that I am not sure how all the functions like :=, order etc.. do work. Does `:=` just work? and all others you would reference explicitly by data.table:: ?

**Review by nikosbosse — CHANGES_REQUESTED** [review]

File: R/stacking_weights_crps.R (hunk 2 above)
Comment (seabbs) [reply R/stacking_weights_crps.R:35]: you need to not import the whole thing. you can import `:=` explicitly. I suggest review carefully.

Comment (seabbs) [reply R/stacking_weights_crps.R:35]: its not used in the function so no.

Comment (seabbs) [reply R/stacking_weights_crps.R:35]: yes you can import `:=` etc explicitly - good idea to check through.

Comment (nikosbosse) [reply R/stacking_weights_crps.R:35]: you added rbindlist in your code suggestions ;)

Comment (seabbs) [reply R/stacking_weights_crps.R:35]: in the example?

Comment (nikosbosse) [reply R/stacking_weights_crps.R:35]: oooh. retracted. thanks :)

File: R/stacking_weights_crps.R [hunk 3]
```diff
@@ -42,20 +42,20 @@
...
+#' testdata <- data[date > splitdate]
 #'
 #' weights <- stackr::crps_weights(traindata)
 #' 
 #' test_mixture <- stackr::mixture_from_samples(testdata, 
 #'                                     weights = weights)
-#' score_df <- rbind(testdata, 
-#'                   test_mixture)
+#' score_df <- data.table::rbindlist(list(testdata, 
+#'                   test_mixture, fill = TRUE)
```
Comment (nikosbosse) [inline R/stacking_weights_crps.R:55]: needs a bracket after test_mixture
and nicer indentation :)

**Review by nikosbosse — CHANGES_REQUESTED** [review]
looks good apart from the bracket, thanks!

**Comment by nikosbosse** [comment]
also just manually changed that. Thanks :)

**Pull request closed without merging**